File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        // dfs, vvector visited, if visited == 0, include in ans vector 
        vector<int>indegree(n, 0);
        
        for(auto e:edges)
            indegree[e[1]]++;
//      0 1 2 3 4 5
//        T T   T  T 
        vector<int>ans;
        for(int i = 0; i < n; i++)
            if(!indegree[i]) ans.push_back(i);
        return ans;
    }
};

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution(object):
    def findSmallestSetOfVertices(self, n, edges):
        """
        :type n: int
        :type edges: List[List[int]]
        :rtype: List[int]
        """
        list1 = [0]*n;
        # print(list1);
        for i in range(len(edges)):
            list1[edges[i][1]] += 1
        list2 = [];
        for i in range(0, n):
            if(list1[i]==0):
                list2.append(i)
        return list2;
            
        